parlai/scripts/train_model.py

```diff
@@ -656,19 +650,43 @@ def _run_eval(
         # write to file
         if write_log and opt.get('model_file') and is_primary_worker():
             # Write out metrics
-            if opt["write_log_as_json"]:
-                with PathManager.open(
-                    opt['model_file'] + log_suffix + '.' + datatype + ".json", 'a'
-                ) as f:
-                    json.dump(dict_report(report), f)
-            else:
-                with PathManager.open(
-                    opt['model_file'] + log_suffix + '.' + datatype, 'a'
-                ) as f:
-                    f.write(f'{metrics}\n')
+            with PathManager.open(
+                opt['model_file'] + log_suffix + '.' + datatype + ".json", 'a'
+            ) as f:
+                json.dump(dict_report(report), f)
```

Comment:
I think we'd prefer keeping it with the rest of the `.trainstats`?

Reply:
The other valid/reports aren't saved to .trainstats right now.

Reply:
Every validation run, except the final one, is stored in trainstats.

https://github.com/facebookresearch/ParlAI/blob/9b5f8c670d21964ed140683a4e3fd1ae7e1f5db3/parlai/scripts/train_model.py#L474-L481

I'm humbly requesting you expand it to also include the final valid/test reports. (They can just be None during the intermediate stages)

Reply:
Alright, wording wasn't quite clear there. Anyway, done.